pkg/volume/glusterfs/glusterfs.go

```diff
@@ -809,6 +809,19 @@ func (p *glusterfsVolumeProvisioner) CreateVolume(gid int) (r *v1.GlusterfsPersi
 	customVolumeName := ""
 	epServiceName := ""
 
+	if len(p.provisionerConfig.customEpNamePrefix) == 0 {
+		epServiceName = string(p.options.PVC.UID)
+	} else {
+		epServiceName = p.provisionerConfig.customEpNamePrefix + "-" + string(p.options.PVC.UID)
+	}
+	epNamespace := p.options.PVC.Namespace
+	endpoint, service, err := p.createEndpointService(epNamespace, epServiceName, p.options.PVC)
```

Comment:
rename createEndpointService to createOrGetEndpointService. since you're using the returned objects to do updates, if they already exist createEndpointService needs to fetch the current versions so we have the current resourceVersion and uid

Reply:
Done.. thanks!